packages/contracts-bedrock/scripts/ChainAssertions.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ProxyAdmin } from "src/universal/ProxyAdmin.sol";
+import { ResourceMetering } from "src/L1/ResourceMetering.sol";
+import { DeployConfig } from "scripts/DeployConfig.s.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+import { Constants } from "src/libraries/Constants.sol";
+import { L1StandardBridge } from "src/L1/L1StandardBridge.sol";
+import { L2OutputOracle } from "src/L1/L2OutputOracle.sol";
+import { ProtocolVersion, ProtocolVersions } from "src/L1/ProtocolVersions.sol";
+import { OptimismPortal } from "src/L1/OptimismPortal.sol";
+import { L1CrossDomainMessenger } from "src/L1/L1CrossDomainMessenger.sol";
+import { OptimismMintableERC20Factory } from "src/universal/OptimismMintableERC20Factory.sol";
+import { L1ERC721Bridge } from "src/L1/L1ERC721Bridge.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { Types } from "scripts/Types.sol";
+import { Vm } from "forge-std/Vm.sol";
+import { ISystemConfigV0 } from "scripts/interfaces/ISystemConfigV0.sol";
+
+library ChainAssertions {
+    /// @notice Asserts the correctness of an L1 deployment
+    function postDeployAssertions(
+        address systemConfig,
+        address protocolVersions,
+        DeployConfig cfg,
+        uint256 l2OutputOracleStartingTimestamp,
+        Vm vm
+    )
+        internal
+        view
+    {
+        SystemConfig sys = SystemConfig(systemConfig);
+        Types.ContractSet memory proxies = Types.ContractSet({
+            L1CrossDomainMessenger: sys.l1CrossDomainMessenger(),
+            L1StandardBridge: sys.l1StandardBridge(),
+            L2OutputOracle: sys.l2OutputOracle(),
```

Comment:
https://github.com/ethereum-optimism/optimism/pull/8094 is going to remove the getter interface for these addresses from the `SystemConfig` so this function will need to be deleted as part of that PR